src/resmom/stage_func.c

```diff
@@ -2284,30 +2329,39 @@ recv_rq_cpyfile_cred(struct rq_cpyfile *pcf)
 
 			if (fgets(buf, sizeof(buf), stdin) == NULL) {
 				free_rq_cpyfile_cred(pcf);
+				log_err(-1, __func__, "Failed to read cred_length");
 				return 0;
 			} else {
 				buf[strlen(buf)-1] = '\0';
 				cred_len = atoi(buf);
 			}
 
-			cred_buf = (char *)malloc(cred_len+1);
-			if (cred_buf == NULL) {
+			if (fgets(buf, sizeof(buf), stdin) == NULL) {
 				free_rq_cpyfile_cred(pcf);
+				log_err(-1, __func__, "Failed to read cred_buf");
 				return 0;
+			} else {
+				buf[strlen(buf)-1] = '\0';
+				cred_buf = NULL;
+				if (decode_from_base64(buf, &cred_buf, &a_cnt)) {
+					sprintf(log_buffer, "Failed to decode buf: %s, len: %d", buf, strlen(buf));
+					log_err(-1, __func__, log_buffer);
+					return 0;
+				}
 			}
 
-			a_cnt = fread(cred_buf, sizeof(char), cred_len+1, stdin);
-			cred_buf[cred_len+1] = '\0';
-			if (cred_len+1 != a_cnt) {
+			if (cred_len != a_cnt) {
+				sprintf(log_buffer, "Mismatch: cred_len = %d, a_cnt=%d", cred_len, a_cnt);
```

Comment:
Can we not use 'a_cnt' and instead mention that we are referring to decoded cred length? I won't know what a_cnt is just from the logs unless I go through the code.